test/libsolidity/semanticTests/inlineAssembly/transient_storage_selfdestruct.sol

```diff
@@ -0,0 +1,52 @@
+contract C {
+    uint8 x;
+
+    constructor() {
+        x = 8;
+        assembly {
+            tstore(0, 13)
+        }
+    }
+
+    function getTransientStorage() external view returns (uint y) {
+        assembly {
+            y := tload(0)
+        }
+    }
+
+    function getX() external view returns (uint8) {
+        return x;
+    }
+
+    function terminate() external {
+        selfdestruct(payable(msg.sender));
+        assert(false);
+    }
+}
+
+contract D {
+    C public c;
+
+    constructor() {
+        c = new C();
+    }
+
+    function f() external {
+        c.terminate();
+        assert(c.getX() == 8);
+        assert(c.getTransientStorage() == 0);
+    }
```

Comment:
You should put something in transient storage before you call `terminate()`. Otherwise there's nothing to clear.

But anyway, from `g()` I already see that this behaves differently than I expected. Neither storage nor transient storage get cleared immediately. I assume storage in `g()` will get cleared after the transaction in, though we can't really test that (because after the transaction the contract won't exist either so we can't call it anymore). Since transient storage is supposed to behave like storage with regard to `selfdestruct`, it is probably not cleared explicitly at all because it only exists for the duration of the transaction.

Reply:
Did you remove this test? We're not marking `selfdestruct` has having effect on transient storage but this test was still useful to confirm that.

Reply:
Yes, I removed it. I will add it again.

Reply:
Done.